### PR TITLE
arm64: bitwise, bit shift, boolean SIMD instructions

### PR DESCRIPTION
**Pull request by mathetake**
This implements bit shit, bitwise and boolean SIMD instructions for
arm64 backend.

Notably, now the impl passes `simd_bitwise.json` `simd_boolean.json`, 
and `simd_bit_shift` spectests.

Signed-off-by: Takeshi Yoneda <takeshi@tetrate.io>

part of #484

### CODE REVIEW

**Comment by mathetake** [comment]
ok the left is boolean instructions!

File: internal/integration_test/asm/arm64_debug/impl_test.go [hunk 1]
```diff
@@ -1276,374 +1275,6 @@ func TestAssemblerImpl_EncodeTwoSIMDBytesToSIMDByteRegister(t *testing.T) {
 	}
 }
 
-func TestAssemblerImpl_EncodeVectorRegisterToVectorRegister(t *testing.T) {
```
Comment (mathetake) [inline internal/integration_test/asm/arm64_debug/impl_test.go:1279]: these are migrated to asm/arm64/impl_tes.go

File: Makefile [hunk 2]
```diff
@@ -117,10 +117,10 @@ spectest:
 	@$(MAKE) spectest.v2
 
 spectest.v1:
-	go test $$(go list ./... | grep $(spectest_v1_dir)) -v -timeout 120s
+	@go test $$(go list ./... | grep $(spectest_v1_dir)) -timeout 120s
 
 spectest.v2:
-	go test $$(go list ./... | grep $(spectest_v2_dir)) -v -timeout 120s
```
Comment (mathetake) [inline Makefile:123]: removed `-v` as it was annoying to see 50,000 OK cases printed on the screen :D

Comment (codefromthecrypt) [reply Makefile:123]: yep we also have that TODO to reduce cardinality at some point..

File: internal/asm/arm64/consts.go [hunk 3]
```diff
@@ -467,32 +471,32 @@ const (
...
-	BLT
-	// BMI is the B.cond instruction with CondMI. https://developer.arm.com/documentation/dui0802/a/A64-General-Instructions/B-cond
-	BMI
-	// BPL is the B.cond instruction with CondPL. https://developer.arm.com/documentation/dui0802/a/A64-General-Instructions/B-cond
-	BPL
-	// BNE is the B.cond instruction with CondNE. https://developer.arm.com/documentation/dui0802/a/A64-General-Instructions/B-cond
-	BNE
-	// BVS is the B.cond instruction with CondVS. https://developer.arm.com/documentation/dui0802/a/A64-General-Instructions/B-cond
-	BVS
+	// BCONDEQ is the B.cond instruction with CondEQ. https://developer.arm.com/documentation/dui0802/a/A64-General-Instructions/B-cond
```
Comment (codefromthecrypt) [inline internal/asm/arm64/consts.go:474]: I think we can re-use the preface docs, then add a link for where the conditions are listed

```suggestion

	// Below are B.cond instructions.
	// 	* https://developer.arm.com/documentation/dui0802/a/A64-General-Instructions/B-cond
	// 	* https://developer.arm.com/documentation/dui0802/a/A32-and-T32-Instructions/Condition-codes

	// BCONDEQ is the B.cond instruction with CondEQ.
```

File: internal/asm/arm64/consts.go [hunk 4]
```diff
@@ -705,28 +710,62 @@ const (
...
 	UMINV
 	// CMEQ is the CMEQ(vector, register) instruction. https://developer.arm.com/documentation/dui0801/g/A64-SIMD-Vector-Instructions/CMEQ--vector--register-
 	CMEQ
-	// ADDP is the ADDP(vector) instruction. https://developer.arm.com/documentation/dui0801/g/A64-SIMD-Vector-Instructions/ADDP--vector-
+	// CMEQZERO is the CMEP(zero) instruction https://developer.arm.com/documentation/ddi0596/2021-12/SIMD-FP-Instructions/CMEQ--zero---Compare-bitwise-Equal-to-zero--vector--?lang=en
+	CMEQZERO
+	// ADDP is the ADDP(scalar) instruction. https://developer.arm.com/documentation/ddi0596/2021-12/SIMD-FP-Instructions/ADDP--scalar---Add-Pair-of-elements--scalar--?lang=en
 	ADDP
+	// VADDP is the ADDP(vector) instruction. https://developer.arm.com/documentation/dui0801/g/A64-SIMD-Vector-Instructions/ADDP--vector-
+	// Note: prefixed by V to distinguish from the non-vector variant of ADDP(scalar).
```
Comment (codefromthecrypt) [inline internal/asm/arm64/consts.go:740]: 👍

File: internal/asm/arm64/impl_test.go [hunk 5]
```diff
@@ -1176,6 +1379,135 @@ func TestAssemblerImpl_EncodeVectorRegisterToVectorRegister(t *testing.T) {
...
+		{
+			name: "and v10.16b, v10.16b, v2.16b",
+			x1:   RegV2,
+			x2:   RegV10,
+			inst: VAND,
+			arr:  VectorArrangement16B,
+			exp:  []byte{0x4a, 0x1d, 0x22, 0x4e, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0},
+		},
+		{
+			// mvn is an alias of NOT: https://developer.arm.com/documentation/ddi0596/2020-12/SIMD-FP-Instructions/MVN--Bitwise-NOT--vector---an-alias-of-NOT-?lang=en
```
Comment (codefromthecrypt) [inline internal/asm/arm64/impl_test.go:1455]: guessed my q

**Review by codefromthecrypt — APPROVED** [review]
Looks good. Maybe check if impl_vec_arm64_test.go is missing any tests of significance. Otherwise, I guess it is up to the spec_test to notice.

**Pull request merged**